adhocracy4/comments_async/serializers.py

```diff
@@ -47,6 +50,9 @@ def to_representation(self, instance):
                 else:
                     categories[category] = category
         ret['comment_categories'] = categories
+        is_blocked = ret.get('is_blocked')
```

Comment:
While in the docs of the model serializer, it says to use this to add fields https://www.django-rest-framework.org/api-guide/serializers/#advanced-serializer-usage
There is also another way of doing that: https://www.django-rest-framework.org/api-guide/relations/#custom-relational-fields
Should we change that or keep it? What do you think @Rineee ?

Reply:
Will wait for @Rineee to feedback on this

Reply:
So you probably thought about that, but what was the problem with just adding `comment = serializers.SerializerMethodField()` and then return an empty string in `get_comment()` if the comment is_blocked?

If that doesnt work, I think we need to leave it here in to_representation as it needs the is_blocked to decide what is serialized And that doesnt work with the above approach as this only gets the comment value from database and then operates on that. So it doesnt have the whole instance (with is_blocked), but just the field itself. I think?

BUT: I would definitely refactor the category serialization above with custom relational fields. I also did that for labels in the idea serializer in Aplus here: https://github.com/liqd/adhocracy-plus/blob/a4d62b89cbe297202a3d18dbb2b9f86ad75eaf5d/apps/ideas/serializers.py#L12.

Reply:
@Rineee adding the SerializerMethodField and get_comment() caused the errors that I initially mentioned in https://github.com/liqd/adhocracy4/pull/885#issue-1047277641, which is why I then edited to_representation to include changing the way the comment text was **represented** instead of overwriting the content of the text permanently (as well as the child-comments) and causing those errors. This approach was visible before I squashed the commits.

Just to make sure I understand your last suggestion correctly: The approach in https://github.com/liqd/adhocracy-plus/blob/a4d62b89cbe297202a3d18dbb2b9f86ad75eaf5d/apps/ideas/serializers.py#L12 to refactor the category serialization would then mean that I 

- take this out of to_representation in the CommentSerializer and move it to a separate class CategorySerializer(serializers.StringRelatedField)?
- Why not model it on the CategoryField(serializers.Field)?

Thanks for your feedback:pray:

Reply:
@Rineee Just to make sure I understand your last suggestion correctly: The approach in https://github.com/liqd/adhocracy-plus/blob/a4d62b89cbe297202a3d18dbb2b9f86ad75eaf5d/apps/ideas/serializers.py#L12 to refactor the category serialization would then mean that I 

- take this out of to_representation in the CommentSerializer and move it to a separate class CategorySerializer(serializers.StringRelatedField)?
- Why not model it on the CategoryField(serializers.Field)?

Thanks for your feedback:pray: